### PR TITLE
[Fix #8788] Change `Style/Documentation` to not trigger offense with only macros

### LINKED ISSUE #8788: Style/Documentation with only macros
I'd like Style/Documentation to accept the following:

```ruby
module RuboCop
  module AST
    # Responsible for compatibility with main gem
    # @api private
    module RuboCopCompatibility
      # ...
    end

    extend RuboCopCompatibility  # => This triggers Style/Documentation: Missing top-level module documentation comment
  end
end
```

I think that (at least) macro calls like `extend` should not trigger it.

### PR DESCRIPTION
**Pull request by tejasbubane**
Closes https://github.com/rubocop-hq/rubocop/issues/8788

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Review by marcandre — APPROVED** [review]

**Comment by tejasbubane** [comment]
How do I generate [expected.html](https://github.com/rubocop-hq/rubocop/blob/c0b8da931e3ec350c1831f2846982249981560af/spec/fixtures/html_formatter/expected.html)? I don't see any rake task and my git-fu isn't helping.

**Comment by marcandre** [comment]
> How do I generate [expected.html](https://github.com/rubocop-hq/rubocop/blob/c0b8da931e3ec350c1831f2846982249981560af/spec/fixtures/html_formatter/expected.html)? I don't see any rake task and my git-fu isn't helping.

You can uncomment the right line in `formatter/html_formatter_spec.rb` (look for `FileUtils.copy`) and run the test...

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,9 @@
 
 * [#8646](https://github.com/rubocop-hq/rubocop/issues/8646): Faster find of all files in `TargetFinder` class which improves initial startup speed. ([@tleish][])
 * [#8102](https://github.com/rubocop-hq/rubocop/issues/8102): Consider class length instead of block length for `Struct.new`. ([@tejasbubane][])
+* [#8646](https://github.com/rubocop-hq/rubocop/issues/8646): Faster find of all files in `TargetFinder` class which improves rubocop initial startup speed. ([@tleish][])
+* [#8102](https://github.com/rubocop-hq/rubocop/issues/8102): Consider class-length instead of block-length for `Struct.new`. ([@tejasbubane][])
+* [#8788](https://github.com/rubocop-hq/rubocop/issues/8788): Change `Style/Documentation` to not trigger offense with only macros. ([@tejasbubane][])
```
Comment (koic) [inline CHANGELOG.md:21]: Can you rebase with the latest master branch and update the changelog?

**Comment by bbatsov** [comment]
Such a change should also be reflected in the cop's documentation.

**Comment by bbatsov** [comment]
@tejasbubane ping :-)

**Comment by tejasbubane** [comment]
Updated changelog and added to cop documentation. Sorry for the delay.

**Comment by koic** [comment]
Thanks!

**Comment by ghiculescu** [comment]
I think this change is a bit too aggressive. For example, this is considered "only macros" so there's no lint error if documentation is missing:

```ruby
class Thing < ApplicationRecord
  belongs_to :user
end
```

My reading of the issue is this is specifically trying to find classes or modules that exclusively `extend` or `include` other modules. Would you be open to making the check more precise?

**Pull request merged**